### PR TITLE
feat: MySQL SSH tunnel 

### LINKED ISSUE #26252: [Feature] MySQL SSH tunnel: Update connection create flow with SSH tunnel context

### PR DESCRIPTION
**Pull request by sumitsum**
## Description
- This PR introduces all the changes required for SSH tunnel to work with MySQL. 
  - create SSH connection as part of datasource create method
  - destroy SSH connection as part of datasource destroy method 
  - validate SSH configuration as part of datasource configuration validation
  - check SSH connection as part of overall connection check before making query
  - update datasource form to add fields for SSH configuration 
- Datasource config for MySQL datasources that have been created so far will be updated via a separate [migration PR](https://github.com/appsmithorg/appsmith/pull/26687) to have the `Standard` option as pre-selected for `Connection method` toggle. 
  - Migration change is separated out in a different PR because a dry run needs to be done against [test](https://www.notion.so/appsmith/Taking-prod-database-snapshot-in-atlas-d53c8b71ddef43e3bbf8e7a362fc8ffe) database to see how much time it would take and if it would go through without any issues. 
  - Older datasource configurations will continue to work as expected even without the migration PR, however, it would not show any value as pre-selected on the `Connection method` toggle. 
- Test cases will raise separately. Tracking it via https://github.com/appsmithorg/appsmith/issues/26254 

https://github.com/appsmithorg/appsmith/assets/1757421/ce4cce0f-05d7-4a62-ba97-c76a55b48d2c

#### PR fixes following issue(s)
Fixes #26252 #26253 #26255

#### Type of change
- New feature (non-breaking change which adds functionality)

#### How Has This Been Tested?
- [x] Manual

#### Dev activity
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

#### QA activity:
- [ ] [Speedbreak features](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#speedbreakers-) have been covered
- [ ] Test plan covers all impacted features and [areas of interest](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#areas-of-interest-)
- [ ] Test plan has been peer reviewed by project stakeholders and other QA members
- [ ] Manually tested functionality on DP
- [ ] We had an implementation alignment call with stakeholders post QA Round 2
- [ ] Cypress test cases have been added and approved by SDET/manual QA
- [ ] Added `Test Plan Approved` label after Cypress tests were reviewed
- [ ] Added `Test Plan Approved` label after JUnit tests were reviewed

### CODE REVIEW

File: app/server/appsmith-plugins/mysqlPlugin/src/main/java/com/external/plugins/MySqlPlugin.java [hunk 1]
```diff
@@ -86,13 +89,15 @@
...
 import static java.nio.charset.StandardCharsets.UTF_8;
+import static org.springframework.util.CollectionUtils.isEmpty;
 
 @Slf4j
 public class MySqlPlugin extends BasePlugin {
 
     private static final int VALIDATION_CHECK_TIMEOUT = 4; // seconds
     private static final String IS_KEY = "is";
     public static final String JSON_DB_TYPE = "JSON";
+    public static final int CONNECTION_METHOD_INDEX = 1;
```
Comment (tkAppsmith) [inline app/server/appsmith-plugins/mysqlPlugin/src/main/java/com/external/plugins/MySqlPlugin.java:100]: what is ensuring this property to be populated at this index everytime??

Comment (sumitsum) [reply app/server/appsmith-plugins/mysqlPlugin/src/main/java/com/external/plugins/MySqlPlugin.java:100]: Hi @tkAppsmith , this index is hardcoded in the `form.json` file which defines the mapping of data. To be specific, this line in `form.json` ensures that the index would always be `1` :
`"configProperty": "datasourceConfiguration.properties[1].value"`

**Review by tkAppsmith — APPROVED** [review]

**Review by tkAppsmith — APPROVED** [review]

**Pull request merged**